### PR TITLE
Python 3.7 tox and travis tests

### PR DESCRIPTION
**Pull request by azmeuk**
This PR adds tox and travis tests for python 3.7, and the setuptools classifier.
It also brings python 3.8 tests for travis.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,9 @@
-os: linux
-sudo: false
```
Comment (davidism) [inline .travis.yml]: Do not remove these.

File: .travis.yml [hunk 2]
```diff
@@ -1,7 +1,9 @@
...
 language: python
+dist: xenial
```
Comment (davidism) [inline .travis.yml]: Move this up with the other os-level spec.

File: .travis.yml [hunk 3]
```diff
@@ -17,6 +19,7 @@ matrix:
   allow_failures:
     - python: nightly
     - python: pypy3
+    - python: 3.8-dev
```
Comment (davidism) [inline .travis.yml]: This is essentially the same as nightly.

**Comment by azmeuk** [comment]
Everything is fixed.

**Comment by whb07** [comment]
looks good to me, @azmeuk great job on helping and quickly fixing the issues brought on by @davidism. Seeing as the changes have passed on Travis, it seems helpful to start having 3.7 tested 
👍

**Pull request merged**